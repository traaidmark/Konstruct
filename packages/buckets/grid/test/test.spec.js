// BUCKET [ GRID > TESTS ] #####################################################

// 1. DEPENDENCIES +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 1.1. EXTERNAL DEPENDENCIES ..................................................

import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

// 1.1. END ....................................................................

// 1.2. INTERNAL DEPENDENCIES ..................................................

import Props from './props';

import Grid from '../src';

// 1.2. END ....................................................................

// 1. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 3. TEST +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

describe('Bucket: Grid', () => {

  // 3.1. RENDER WITHOUT PROPS .................................................

  test('renders correctly without any props', () => {
    const tree = renderer.create(
      <Grid>
        <div>col 1</div>  
        <div>col 2</div>  
      </Grid>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // 3.1. END ..................................................................

  // 3.2. RENDER WITH ALL PROPS ................................................

  test('renders correctly with all props', () => {
    const tree = renderer.create(
      <Grid { ...Props }>
        <div>col 1</div>  
        <div>col 2</div>  
      </Grid>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // 3.2. END ..................................................................

});

// 3. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// END OF FILE #################################################################
