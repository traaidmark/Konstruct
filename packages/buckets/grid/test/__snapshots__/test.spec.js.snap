// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bucket: Grid renders correctly with all props 1`] = `
<div
  className="b-grid "
  style={
    Object {
      "--grid-col-l": "1fr 1fr",
      "--grid-col-m": "1fr 1fr",
      "--grid-col-s": "1fr 1fr",
      "--grid-col-xl": "1fr 1fr",
      "--grid-gap-l": "var(--gutter-medium)",
      "--grid-gap-m": "var(--gutter-medium)",
      "--grid-gap-s": "var(--gutter-medium)",
      "--grid-gap-xl": "var(--gutter-medium)",
    }
  }
>
  <div>
    col 1
  </div>
  <div>
    col 2
  </div>
</div>
`;

exports[`Bucket: Grid renders correctly without any props 1`] = `
<div
  className="b-grid "
  style={
    Object {
      "--grid-col-l": undefined,
      "--grid-col-m": undefined,
      "--grid-col-s": undefined,
      "--grid-col-xl": undefined,
      "--grid-gap-l": undefined,
      "--grid-gap-m": undefined,
      "--grid-gap-s": undefined,
      "--grid-gap-xl": undefined,
    }
  }
>
  <div>
    col 1
  </div>
  <div>
    col 2
  </div>
</div>
`;
